deflate_medium.c

```
@@ -23,31 +23,32 @@ struct match {
  * - s->lookahead > match.match_length + WANT_MIN_MATCH
  * - match_len >= WANT_MIN_MATCH
+ * - max_len == s->max_insert_length * 16
  */
 static void insert_match(deflate_state *s, unsigned char *Z_RESTRICT window, struct match match, const uint32_t max_len) {
+    uint32_t start;
     uint32_t match_len = match.match_length;
-    uint32_t strstart = match.strstart;
+    uint32_t strstart = match.strstart + 1; // string at strstart already in table
+    uint32_t end = strstart + match_len - 1;
 
     /* Insert new strings in the hash table only if the match length
```

Cppcheck (2.21.0)

[error] `#error` MAX_WBITS must be in 9..15

(preprocessorErrorDirective)